packages/builder/src/steps/contract.ts

```diff
@@ -108,6 +82,8 @@ export default {
   },
 
   configInject(ctx: ChainBuilderContextWithHelpers, config: Config) {
+    validateConfig('contract', config);
```

Comment:
rather than verifying the config at the beginning here why not change the `ChainDefinition` class to automatically validate before calling config inject? otherwise, this is likely to be missed by plugin integrators.